astropy/wcs/wcs.py

```diff
@@ -3076,7 +3076,9 @@ def pixel_scale_matrix(self):
         except InconsistentAxisTypesError:
             try:
                 # for non-celestial axes, get_cdelt doesn't work
-                cdelt = np.dot(self.wcs.cd, np.diag(self.wcs.cdelt))
+                with warnings.catch_warnings():
+                    warnings.filterwarnings('ignore', 'cdelt will be ignored since cd is present', RuntimeWarning)
+                    cdelt = np.dot(self.wcs.cd, np.diag(self.wcs.cdelt))
```

Comment:
I don't understand this line. `InconsistentAxisTypesError` is raised when there are non-celestial axes.
Why is it assumed that a `CD` is defined in this case? And when a `CD` is present `CDELT` is always (1, 1) so there's no reason to include it in any calculations. But more importantly how does the above line compute a scaling factor when it includes the rotation as well. Perhaps I misunderstand the intention here.

Reply:
@keflavich - as the author of this function, could you chime in here regarding @nden's comment?

Reply:
uh oh.  Um... You're certainly talking to a different @keflavich than the one that wrote that n years ago.

I don't know.  I always used to have headaches when accessing multiple FITS files whose WCS attributes I didn't know a priori - if they were a mix of `CD`, `CDELT`, and `PC`-based headers - so this might have been intended as a way to catch a case where I ask for `cdelt` but `cd` is defined?  I'm really not sure, though =(

Reply:
@astrofrog I suggest we merge this PR and file an issue to fix the function?

Reply:
@nden - sounds good to me. Could you please then hit the merge button when you think it's ready?